x-pack/plugins/canvas/public/state/selectors/workpad.js

```diff
@@ -128,8 +128,9 @@ export function getGlobalFilterExpression(state) {
 }
 
 // element getters
-export function getSelectedElementId(state) {
-  return get(state, 'transient.selectedElement');
+function getSelectedElementId(state) {
+  const toplevelNodes = get(state, 'transient.selectedToplevelNodes') || [];
+  return toplevelNodes.length === 1 ? toplevelNodes[0] : null;
```

Comment:
Q: If I read this right, a selected group will return null/no selected elements, right? And I assume that so far you have not needed to get selected elements in the group.

Reply:
Correct, for ad-hoc groups, which have to have more than one element (one can't currently form a group of a single element), and they're top level elements, because it's not a proper group yet, just the user selecting a bunch of nodes. For persistent groups, it does convey the selection. So it does more than it used to - in `master`, persistent groups will result a `null`. 

This function is only ever called by `getSelectedElement` which won't find the persistent group among `elements` (because it's a group). I should rename `getSelectedElementId` to `getSelectedNodeId`... it's a bit untied as some of the changes will be basis for side panel functionality, eg. cloning a group or bringing the group to the top or bottom (these are now doable only via a shortcut), and as `getSelectedNodeId` and `getSelectedElement` are considered legacy, they'll both go away.

Even in future PRs, the meaning of `selectedToplevelNodes` will be the things the user can directly select (click on, or make an ad hoc group out of), so in the case of a persistent group, it's the group itself.

Reply:
Oh, so in persisted groups, `toplevelNodes` would be the group, and would thus be returned. Makes sense, thanks.